vlog4j-graal/src/main/java/org/semanticweb/vlog4j/graal/GraalImporter.java

```diff
@@ -0,0 +1,214 @@
+
+import fr.lirmm.graphik.graal.api.core.AtomSet;
+import fr.lirmm.graphik.graal.api.core.ConjunctiveQuery;
+import fr.lirmm.graphik.util.stream.CloseableIterator;
+import fr.lirmm.graphik.util.stream.IteratorException;
+
+/**
+ * Utility class to convert Graal data structures into VLog4J structures.
+ * Labels are not imported since VLog4J does not support them.
+ * @author adrian
+ *
+ */
+public class GraalImporter {
+
+	private GraalImporter() {};
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atom} into a {@link Atom VLog4J Atom}.
+	 * @param atom A {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atom}
+	 * @return A {@link Atom VLog4J Atom}
+	 */
+	public static Atom importAtom(fr.lirmm.graphik.graal.api.core.Atom atom) {
+		Predicate predicate = importPredicate(atom.getPredicate());
+		List<Term> terms = importTerms(atom.getTerms());
+		return makeAtom(predicate, terms);
+	}
+	
+	/**
+	 * Converts a {@link List} of {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atoms} into a
+	 * {@link List} of {@link Atom VLog4J Atoms}.
+	 * @param atoms A {@link List} of {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atoms}.
+	 * @return A {@link List} of {@link Atom VLog4J Atoms}.
+	 */
+	public static List<Atom> importAtoms(List<fr.lirmm.graphik.graal.api.core.Atom> atoms) {
+		List<Atom> result = new ArrayList<>();
+		for (fr.lirmm.graphik.graal.api.core.Atom atom : atoms) {
+			result.add(importAtom(atom));
+		}
+		return result;
+	}
+
+	/**
+	 * Converts a {@link AtomSet Graal AtomSet} into a {@link Conjunction VLog4J Conjunction}.
+	 * @param atomSet A {@link AtomSet Graal Atomset}
+	 * @return A {@link Conjunction VLog4J Conjunction}
+	 */
+	private static Conjunction importAtomSet(AtomSet atomSet) {
+		List<Atom> result = new ArrayList<>();
+		try (CloseableIterator<fr.lirmm.graphik.graal.api.core.Atom> iterator = atomSet.iterator()) {
+			while (iterator.hasNext())
+				result.add(importAtom(iterator.next()));
+		} catch (IteratorException e) {
+			throw new GraalImportException(MessageFormat.format("Unexpected Iterator Exception when importing AtomSet {0}}.", atomSet));
+		}
+		return makeConjunction(result);
+	}
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Constant Graal Constant} into a {@link Constant VLog4J Constant}.
+	 * @param constant A {@link fr.lirmm.graphik.graal.api.core.Constant Graal Constant}
+	 * @return A {@link Constant VLog4J Constant}
+	 */
+	private static Constant importConstant(fr.lirmm.graphik.graal.api.core.Constant constant) {
+		return makeConstant(constant.getIdentifier().toString());
+	}
+	
+	/**
+	 * Converts a {@link Set} of {@link fr.lirmm.graphik.graal.api.core.Constant Graal Constants} into a {@link Set} of {@link Constant VLog4J Constants}.
+	 * @param constants {@link Set} of {@link fr.lirmm.graphik.graal.api.core.Constant Graal Constants}
+	 * @return {@link Set} of {@link Constant VLog4J Constants}
+	 */
+	@SuppressWarnings("unused")
```

Comment:
private method is unused. If it not planned to be used by further Github issues, it is best to remove it, for a cleaner and more maintainable code.

Reply:
Removed unused code.